### DISCUSSION TITLE
Can I use authentik to create a Signup/Signin with email and password (with no external services)?

### DISCUSSION THREAD

**Discussion by frederikhors** [goauthentik/authentik#1457] [Q&A]
I just found this amazing project. Thanks for your precious work here.

I'm wondering if this is similar to https://github.com/ory/kratos.

Can I use authentik to create a Signup/Signin with email and password (with no external services)?

**Comment by BeryJu** [comment]
Hi, thanks!

Yes, you can use authentik for that, including MFA and all other sorts of custom logic.

**Reply by frederikhors** [reply]
I cannot find any docs on this. I need to use authentik in an offline LAN project. I don't need any third-party services. How to?

And also, can I embed authetnik in a Golang project instead of calling it with API?

**Reply by ser** [reply]
> Hi, thanks!
> 
> Yes, you can use authentik for that, including MFA and all other sorts of custom logic.

How? Could you please point to the ready to use application?

**Reply by genofire** [reply]
@BeryJu - how?

**Reply by SkypLabs** [reply]
Logging into Authentik via its web API is documented here: https://goauthentik.io/developer-docs/api/flow-executor.

I'll try to post a PoC here later on.

**Comment by kinthaiofficial** [comment]
Yes, authentik handles email-based signup/signin well with its built-in flows. Here's the setup:

**1. Create an enrollment flow** (signup):
```
Admin → Flows & Stages → Create Flow
Type: Enrollment
Stages to add (in order):
1. Email Stage — collect email
2. Password Stage — set password (optional: use passwordless instead)
3. User Write Stage — create the user account
4. Email Verification Stage — send confirmation email
5. Login Stage — log them in after verification
```

**2. Create an authentication flow** (signin):
```
Stages:
1. Identification Stage (source: email)
2. Password Stage
3. (Optional) MFA Stage for 2FA
```

**3. Wire up via Provider + Application**:
```yaml
# Application in authentik points to your frontend
Launch URL: https://yourapp.com
Provider: OAuth2 or SAML
```

**Minimal working example with the API**:
```python
import requests

# Create a user programmatically (if building custom flows)
response = requests.post(
    "https://authentik.example.com/api/v3/core/users/",
    headers={"Authorization": f"Bearer {api_token}"},
    json={
        "username": "user@example.com",
        "email": "user@example.com",
        "name": "New User",
        "is_active": True,
    }
)
user = response.json()

# Set password
requests.post(
    f"https://authentik.example.com/api/v3/core/users/{user['pk']}/set_password/",
    headers={"Authorization": f"Bearer {api_token}"},
    json={"password": "securepassword123"},
)
```

The flow builder in the Admin UI is genuinely powerful once you understand the stage concept. Would recommend the flow-based approach over the API for anything that needs email verification — it handles edge cases (email bounce, re-verification) that are annoying to implement yourself.
